code/src/components/HappyForm.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react'
+import './HappyForm.css'
+
+const HappyForm = ({ onFormSubmit, newThought, setNewThought }) => {
+
+  return (
+
+      <form className="happyThoughtInput" onSubmit={onFormSubmit}>
+        <label>
+          What's making you happy?
+        <textarea
+          placeholder="Type here :)"
+          row="2"
+          minLength="4"
+          maxLength="140"
+          required
+          value={newThought}
+          onChange={(event) => setNewThought(event.target.value)}
+        />
+        
+        
+
+        <button className="happyThoughtButton" disabled=
+                {newThought.length < 4 || newThought.length > 140}
```

Comment:
I think I only had this part without the min and max length above, do they both fill a function?